Source/HTTP/iOS/http_ios.mm

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) Microsoft Corporation
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+#include "pch.h"
+#include "http_ios.h"
+
+ios_http::ios_http(_In_ AsyncBlock* asyncBlock, _In_ hc_call_handle_t call) :
+    m_call(call),
+    m_asyncBlock(asyncBlock)
+{
+    NSURLSessionConfiguration* configuration = NSURLSessionConfiguration.ephemeralSessionConfiguration;
+
+    uint32_t timeoutInSeconds = 0;
+    if (FAILED(HCHttpCallRequestGetTimeout(m_call, &timeoutInSeconds)))
+    {
+        // default to 60 to match other default ios behaviour
+        timeoutInSeconds = 60;
+    }
+    
+    [configuration setTimeoutIntervalForRequest:(NSTimeInterval)timeoutInSeconds];
+    [configuration setTimeoutIntervalForResource:(NSTimeInterval)timeoutInSeconds];
+    
+    m_session = [NSURLSession sessionWithConfiguration:configuration];
+}
+
+void ios_http::completion_handler(NSData *data, NSURLResponse *response, NSError *error)
+{
+    ASSERT([response isKindOfClass:[NSHTTPURLResponse class]]);
+    NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse*)response;
+    
+    if (error)
+    {
+        // TODO: handle errors
```

Comment:
>TODO [](start = 11, length = 4)

Can we file a task for this?

Reply:
We can, but I'm going to be doing that as soon as this PR completes, so I'm not sure it's worth it?

---
In reply to: [188793811](https://github.com/Microsoft/libHttpClient/pull/162#discussion_r188793811) [](ancestors = 188793811)